File: 3329-find-the-length-of-the-longest-common-prefix/3329-find-the-length-of-the-longest-common-prefix.cpp
class Solution {
public:
    int longestCommonPrefix(vector<int>& arr1, vector<int>& arr2) {
        unordered_set<string>mySet;
        for(auto &num : arr1) {
            string s = to_string(num);
            for(int i = 1 ; i <= s.length(); i++) {
                mySet.insert(s.substr(0,i));
            }
        }
        int cnt = 0;
        for(auto& num : arr2) {
            string s = to_string(num);
            for(int i = 1 ; i <= s.length(); i++) {
                string str = s.substr(0, i);
                if(mySet.contains(s.substr(0, i))) {
                    int len = str.length();
                    cnt = max(cnt, len);
                }
            }
        }
        return cnt;
    }
};

File: 3329-find-the-length-of-the-longest-common-prefix/3329-find-the-length-of-the-longest-common-prefix.java
class Solution {
    public int longestCommonPrefix(int[] arr1, int[] arr2) {
        HashSet<String> mySet = new HashSet<>();
        for(int num : arr1) {
            String str = Integer.toString(num);
            for(int i = 1; i <= str.length(); i++) {
                mySet.add(str.substring(0, i));
            }
        }
        int cnt = 0;
        for(int num : arr2) {
            String str = Integer.toString(num);
            for(int i = 1; i <= str.length(); i++) {
                String myString = str.substring(0, i);
                if(mySet.contains(myString)) {
                    cnt = Integer.max(cnt, myString.length());
                }
            }
        }
        return cnt;

    }
}